src/wizard/index.tsx

```diff
@@ -49,7 +56,15 @@ export default factory(function Wizard({ properties, children, middleware: { the
 		theme: themeProp
 	} = properties();
 
-	const [{ step: stepRenderer } = { step: undefined }] = children();
+	let stepRenderer: StepRenderer | undefined;
+	let stepContent: DNode[] | undefined;
+	const body = children();
+
+	if (body && body[0] && !isRenderResult(body[0])) {
```

Comment:
This is quite confusing, i think it can be a little simplier.

```tsx
const [renderer] = children();

if (renderer && typeof renderer.step === 'function') {
    stepRenderer = renderer.step;
}
```

and then because of the condition in stepWrappers you know you can use `children()[index]` as if it was custom renderer it would have already returned.

I think that covers the two scenarios... I a little bit clearer i think.

Reply:
`renderer.step` still throws in TS; that's why I was using `isRenderResult`. That's the only real difference here imo, but I will try to make it clearer.

Reply:
Cleaned up in 47f4e52.